### ISSUE TITLE
handling newly introduced deprecations (SKIP 4 clarification needed)

### ISSUE THREAD

**Issue by grlee77** [scikit-image/scikit-image#6448]
## Description

We have some existing kwarg deprections that were introduced in 0.19 where the "removed_version" is listed as 1.0, but I think that was only correct assuming that were going to follow SKIP 3. If we follow SKIP 4, though, my understanding is that these would not ever get removed from `skimage`, but would instead be removed in the `skimage2` package. 

We have more than 50 current `deprecate_kwarg` calls with `deprecated_version="0.19", removed_version="1.0" (i.e. the dozens of functions where we are moving from `selem` to `footprint`). I also found one case of removed_version="1.2"! Before releasing 1.0, should we just modify the text of these to be something like removed_version="skimage2"? (I don't think we just want to immediately deprecate in 1.0 as that would be disruptive for deprecations we just introduced in the current release.)

A similar ting came up in the following open PR: https://github.com/scikit-image/scikit-image/pull/6418#discussion_r922828506

attn: @skimage/core

**Comment by lagru** [comment]
> should we just modify the text of these to be something like removed_version="skimage2"?

Targeting `skimage2` for the deprecations sounds like the most reasonable approach to me. However, I feel that just adding `skimage2` might lead to some confusion. Thinking out loud, we could add a new parameter `hint` to the deprecation decorators. That would allow providing more context for this  and other cases as well. Also, technically it should be `removed_version=2.0`. [SKIP 4](https://github.com/scikit-image/scikit-image/blob/e9deba5bff67c9faaf33fe85881fd43f9d693837/doc/source/skips/4-transition-to-v2.rst#implementation) proposes to change the distribution package name and import name to `skimage2` and the version to `2.0`. 

I don't think your summoning worked. Let me try :mage:: @scikit-image/core

**Comment by alexdesiqueira** [comment]
@grlee77 
> I also found one case of removed_version="1.2"! 

:sweat_smile: 

@lagru :mage: :boom: 
> Also, technically it should be removed_version=2.0. [SKIP 4](https://github.com/scikit-image/scikit-image/blob/e9deba5bff67c9faaf33fe85881fd43f9d693837/doc/source/skips/4-transition-to-v2.rst#implementation) proposes to change the distribution package name and import name to skimage2 and the version to 2.0.

Yeah, I think that should be the way to do it. _However_, SKIP 4 doesn't appear as approved yet? What's the status on that one?

**Comment by lagru** [comment]
My feeling is that there is a silent consensus that SKIP 4 is the way to go? So perhaps it's time to kickoff the [process for moving it to Accepted](https://github.com/scikit-image/scikit-image/blob/e9deba5bff67c9faaf33fe85881fd43f9d693837/doc/source/skips/0-skip-process.rst#how-a-skip-becomes-accepted).

**Comment by alexdesiqueira** [comment]
Either that, or we toss it away and approve [SKIP 3](https://scikit-image.org/docs/stable/skips/3-transition-to-v1.html) while they're sleeping... :pirate_flag: :skull_and_crossbones: :joy:

**Comment by alexdesiqueira** [comment]
Ping @scikit-image/core; this is urgent.

**Comment by stefanv** [comment]
Should SKIP4 prevent us from releasing a 1.0?

**Comment by alexdesiqueira** [comment]
@stefanv no, but it will prevent us to release 1.1—according to SKIP 4 as is right now.

**Comment by lagru** [comment]
There is the idea to release a version 0.20 before 1.0 so that we have space for the proper deprecation cycle. That would require amending SKIP 4 a little bit. This could relieve some pressure.

**Comment by stefanv** [comment]
Yes, we shouldn't let the SKIP make regular releases harder, especially since we cannot predict how much time we will have to work on its implementation.

**Comment by jni** [comment]
I need to update the SKIP both to allow more flexible upcoming releases and also to update the top motivation section, as discussed with @stefanv [in the original PR](https://github.com/scikit-image/scikit-image/pull/6339#issuecomment-1109148040). Apologies for the slowness.

**Comment by lagru** [comment]
@jni if you don't have the bandwidth for it I can take a shot at this. In that case you could feel free to view it as a joined PR where you can push changes as you want. :wink:
